lib/net/tns/client.rb

```diff
@@ -58,6 +58,8 @@ def self.get_status(opts={})
             # a data length that is greater than the available data.
           end
 
+          response = conn.send_and_receive(ResendPacket.new())
```

Comment:
Code above (rescue) results in response being nil initially, so adding ResendPacket is just a workaround to allow code not to fail with "undefined method `data' for nil:NilClass (NoMethodError)" on next line. At this point server response (status) is received already so more appropriate solution might be to remove ResendPacket.new() and the next line altogether.